backend/Application/Api/GraphQL/Network/ClientVersionComparer.cs

```diff
@@ -0,0 +1,48 @@
+namespace Application.Api.GraphQL.Network
+{
+    public class ClientVersionComparer : IComparer<string?>
+    {
+        public int Compare(string? x, string? y)
+        {
+            if (String.IsNullOrEmpty(x) && String.IsNullOrEmpty(y))
+            {
+                return 0;
+            }
+            else if (String.IsNullOrEmpty(x) && !String.IsNullOrEmpty(y))
+            {
+                return -1;
+            }
+            else if (!String.IsNullOrEmpty(x) && String.IsNullOrEmpty(y))
+            {
+                return 1;
+            }
+            else if (!String.IsNullOrEmpty(x) && !String.IsNullOrEmpty(y))
+            {
+                var xParts = x.Split(".");
+                var yParts = y.Split(".");
+
+                if (xParts.Length != yParts.Length)
+                {
+                    return 0;
+                }
+
+                for (int i = 0; i < xParts.Length; i++)
+                {
+                    var xPart = Int64.Parse(xParts[i]);
+                    var yPart = Int64.Parse(yParts[i]);
```

Comment:
I am not sure this is safe. What if put a version as `1.2.3-rc1`?

You don't really have any guarantees about the version, and we don't want the backend to raise an exception just because somebody puts a dummy version in the list.

Reply:
we have fixed semver parsing. It should not raise an exception till the version is any valid Semver.